src/software/thunderscope/robot_diagnostics/drive_and_dribbler_widget.py

```diff
@@ -16,32 +24,61 @@ def __init__(self, proto_unix_io: ProtoUnixIO) -> None:
 
         self.proto_unix_io = proto_unix_io
 
-        # Add widgets to layout
-        layout.addWidget(self.setup_direct_velocity("Drive"))
-        layout.addWidget(self.setup_dribbler("Dribbler"))
+        # create swappable widget system using stacked widgets
+        self.direct_velocity_widget = self.setup_direct_velocity(
+            "Drive - Direct Velocity"
+        )
+        self.per_motor_widget = self.setup_per_motor("Drive - Per Motor")
+        self.drive_widget.addWidget(self.direct_velocity_widget)
+        self.drive_widget.addWidget(self.per_motor_widget)
 
-        self.enabled = True
+        layout.addWidget(self.setup_drive_switch_radio("Drive Mode Switch"))
+        layout.addWidget(self.drive_widget)
+        layout.addWidget(self.setup_dribbler("Dribbler"))
 
+        # default to direct velocity
+        self.drive_mode = DriveMode.VELOCITY
+        self.use_direct_velocity.setChecked(True)
+        self.toggle_drive_mode(DriveMode.VELOCITY)
         self.setLayout(layout)
+        self.toggle_dribbler_sliders(True)
 
     def refresh(self) -> None:
         """Refresh the widget and send the a MotorControl message with the current values"""
         motor_control = MotorControl()
         motor_control.dribbler_speed_rpm = int(self.dribbler_speed_rpm_slider.value())
 
-        motor_control.direct_velocity_control.velocity.x_component_meters = (
-            self.x_velocity_slider.value()
-        )
-        motor_control.direct_velocity_control.velocity.y_component_meters = (
-            self.y_velocity_slider.value()
-        )
-        motor_control.direct_velocity_control.angular_velocity.radians_per_second = (
-            self.angular_velocity_slider.value()
-        )
+        if not self.drive_mode == DriveMode.VELOCITY:
+            motor_control.ClearField("direct_per_wheel_control")
+            motor_control.direct_velocity_drive.velocity.x_component_meters = (
+                self.x_velocity_slider.value()
+            )
+            motor_control.direct_velocity_drive.velocity.y_component_meters = (
+                self.y_velocity_slider.value()
+            )
+            motor_control.direct_velocity_drive.angular_velocity.radians_per_second = (
+                self.angular_velocity_slider.value()
+            )
+        else:
+            motor_control.ClearField("direct_velocity_control")
+            motor_control.direct_per_wheel_drive.front_left_wheel_velocity = (
+                self.front_left_motor_slider.value()
+            )
+            motor_control.direct_per_wheel_drive.front_right_wheel_velocity = (
+                self.front_right_motor_slider.value()
+            )
+            motor_control.direct_per_wheel_drive.back_left_wheel_velocity = (
+                self.back_left_motor_slider.value()
+            )
+            motor_control.direct_per_wheel_drive.back_right_wheel_velocity = (
+                self.back_right_motor_slider.value()
+            )
```

Comment:
nit: why structure it this way?

why not `if self.drive_mode == DriveMode.MOTOR` to avoid that confusing `not` if you are handling the `else` case here as well?

so
```
if self.drive_mode == DriveMode.MOTOR
    ...
else:
```